tests/qam-updinstall/update_install.pm

```diff
@@ -86,6 +87,46 @@ sub change_repos_state {
     zypper_call("mr --$state $repos");
 }
 
+sub get_installed_bin_version {
+    my $name = $_[0];
+    if (not script_run("rpm -q $name")) {
+        return script_output "rpm -q --queryformat '%{VERSION}-%{RELEASE}' $name";
+    } else {
+        return 0;
+    }
+}
+
+sub report_results {
```

Comment:
I would not create `report_results` if you don't use it more than once

Reply:
I reworked it to be less verbose and be called once for each type package.